tensorflow/contrib/framework/python/framework/graph_util.py

```diff
@@ -126,3 +128,30 @@ def fuse_op(graph_def, input_nodes, output_nodes, output_dtypes,
   out.library.CopyFrom(graph_def.library)
   out.versions.CopyFrom(graph_def.versions)
   return out
+
+
+def get_placeholders(graph):
+  """Get placeholders of a graph.
+
+  Args:
+    graph: A tf.Graph.
+  Returns:
+    A list contains all placeholders of given graph.
+
+  Raises:
+    TypeError: If 'graph' is not a tensorflow graph.
+  """
+
+  if graph and not isinstance(graph, ops.Graph):
+    raise TypeError("Input graph needs to be a Graph: %s" % graph)
+
+  # For each placeholder() call, there is a coresponding
+  # operation of type 'Placeholder' registered to the graph.
+  # The return value (a Tensor) of placeholder() is the
+  # first output of this operation in fact.
+  operations = graph.get_operations()
+  result = []
+  for i in operations:
+    if i.type == 'Placeholder':
+      result.append(i.outputs[0])
+  return result
```

Comment:
how about `result = [op.outputs[0] for op in operations if op.type == "Placeholder"]`